### PR TITLE
added admin endpoint for global markups management

### PR DESCRIPTION
**Pull request by DFSko**
added admin endpoint for global markups management

resolves happy-travel/agent-app-project#62

### CODE REVIEW

File: Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+using HappyTravel.Edo.Api.Models.Markups;
+using HappyTravel.Edo.Api.Services.Markups;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HappyTravel.Edo.Api.Controllers.AdministratorControllers
+{
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Route("api/{v:apiVersion}/admin/markups/global")]
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs:16]: `global-markups` maybe?
in other places markups are the part of resource (counterparty, agent, agency).
here it looks like independent resource

File: Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+        
+        
+        /// <summary>
+        /// Gets global markup policies
+        /// </summary>
+        /// <returns>List of global markups</returns>
+        [HttpGet]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(List<MarkupInfo>), StatusCodes.Status200OK)]
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs:32]: Fine

File: Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        /// </summary>
+        /// <param name="policyId">Id of the policy to delete.</param>
+        /// <returns></returns>
+        [HttpDelete("{policyId:int}")]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [AdministratorPermissions(AdministratorPermissions.MarkupManagement)]
+        public async Task<IActionResult> RemovePolicy([FromRoute] int policyId)
+        {
+            var (_, isFailure, error) = await _policyManager.RemoveGlobalMarkupPolicy(policyId);
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs:70]: Markup can be omitted there, to be similar with other methods

File: Api/Services/Markups/IAdminMarkupPolicyManager.cs [hunk 4]
```diff
@@ -14,6 +14,14 @@ public interface IAdminMarkupPolicyManager
         Task<Result> Modify(int policyId, MarkupPolicySettings settings);
 
         Task<List<MarkupPolicyData>> Get(MarkupPolicyScope scope);
+        
+        Task<List<MarkupInfo>> GetGlobalMarkupPolicies();
+
+        Task<Result> AddGlobalMarkupPolicy(MarkupPolicySettings settings);
+
+        Task<Result> RemoveGlobalMarkupPolicy(int policyId);
```
Comment (mingazhev) [inline Api/Services/Markups/IAdminMarkupPolicyManager.cs:22]: Let's remove markup word frome these three methods.
Just AddGlobalPolicy and etc.

File: Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs (hunk 1 above)
Comment (DFSko) [reply Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs]: Fixed

File: Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs (hunk 3 above)
Comment (DFSko) [reply Api/Controllers/AdministratorControllers/GlobalMarkupsController.cs]: Fixed

File: Api/Services/Markups/IAdminMarkupPolicyManager.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Markups/IAdminMarkupPolicyManager.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**